website/en/docs/types/functions.md

```diff
@@ -213,6 +213,53 @@ var num: number = method.call(42);
 var str: string = method.call(42);
 ```
 
+### Functions with %checks <a class="toc" id="toc-function-checks" href="#toc-function-checks"></a>
+
+Sometimes you will want to move condition from `if` statement into a function
+
+```js
+function concat(a: ?string, b: ?string): string {
+  if (a && b) {
+    return a + b;
+  }
+  return '';
+}
+```
+
+However, `flow` will see an error in the code below:
+
+```js
+function truthy(a, b): boolean {
+  return a && b;
+}
+
+function concat(a: ?string, b: ?string): string {
+  if (truthy(a, b)) {
+    // $ExpectError
+    return a + b;
+  }
+  return '';
+}
+```
+
+You can fix it using `%checks` mark like this:
+
+```js
+function truthy(a, b): boolean %checks {
+  return !!a && !!b;
+}
+
+function concat(a: ?string, b: ?string): string {
+  if (truthy(a, b)) {
+    return a + b;
+  }
+  return '';
+}
+```
+
+**You can't move such function with `%checks` to another file currently.**
```

Comment:
Moving `truthy` to a separate file and keeping the intended functionality is possible. You would, however, have  to include type annotations for the arguments, e.g.
```
export function truthy(a: mixed, b: mixed): boolean %checks {
  return !!a && !!b;
}
```